src/client/address_book/storage.cc

```diff
@@ -68,15 +69,20 @@ bool AddressBookStorage::GetAddress(
 void AddressBookStorage::AddAddress(
     const kovri::core::IdentityEx& address) {
   auto filename = GetAddressesPath() / (address.GetIdentHash().ToBase32() + ".b32");
-  std::ofstream file(filename.string(), std::ofstream::binary);
-  if (!file)
+  kovri::core::OutputFileStream file(filename.string(), std::ofstream::binary);
+  if (file.Bad() || file.Fail())
     LOG(error) << "AddressBookStorage: can't open file " << filename;
   const std::size_t len = address.GetFullLen();
   auto buf = std::make_unique<std::uint8_t[]>(len);
   // For sanity, the validity of identity length is incumbent upon the parent caller.
   // TODO(unassigned): triple check that this is the case
   address.ToBuffer(buf.get(), len);
-  file.write(reinterpret_cast<char *>(buf.get()), len);
+  if (!file.Write(buf.get(), len))
+    {
+      core::Exception ex;
```

Comment:
This should be a member to `AddressBookStorage`. The class acts on enough data to warrant it's own exception member.